src/clients/dotnet/TigerBeetle/UInt128Extensions.cs

```diff
@@ -101,3 +102,65 @@ public static UInt128 ToUInt128(this BigInteger value)
         }
     }
 }
+
+/// <summary>
+/// Safe generation of Universally Unique and Byte-Sortable Identifiers as UInt128s.
+/// </summary>
+public static class ID
+{
+    private static long idLastTimestamp = 0L;
+    private static readonly byte[] idLastRandom = new byte[12];
+
+    /// <summary>
+    /// Generates a universally unique identifier as a UInt128.
+    /// Each call generates a monotonically increasing value to last under sequential consistency.
+    /// This function is thread-safe.
+    /// </summary>
+    public static UInt128 Create()
+    {
+        // Safe to cast from long to ulong as, in C#, this
+        var timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+        ulong randomLo;
+        ushort randomHi;
+
+        lock (idLastRandom)
+        {
+            if (timestamp <= idLastTimestamp)
+            {
+                timestamp = idLastTimestamp;
+            }
+            else
+            {
+                idLastTimestamp = timestamp;
+                RandomNumberGenerator.Fill(idLastRandom);
+            }
+
+            Span<byte> lastRandom = idLastRandom;
+            randomLo = BitConverter.ToUInt64(lastRandom.Slice(0));
+            randomHi = BitConverter.ToUInt16(lastRandom.Slice(8));
+
+            // Increment the u80 stored in lastRandom using a u64 increment then u16 increment.
+            // Throws an exception if the entire u80 represented with both overflows.
+            // In C#, unsigned arithmetic wraps around on overflow by default so check for zero.
+            randomLo += 1;
+            if (randomLo == 0)
+            {
+                randomHi += 1;
```

Comment:
Although the default behavior in C# is unchecked arithmetic, it can be changed by a compiler flag.
Let's make it explicit that we rely on overflow here.

```suggestion
                unchecked
                {
                    randomHi += 1;
                }
```

Reply:
TIL about the compiler flag. Just to _check_, `unchecked` around both increments right? (as they rely on overflow)